src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -214,6 +236,44 @@ public virtual void ProcessBlock(Block block)
             this.logger.LogTrace("(-)");
         }
 
+        /// <inheritdoc />
+        public virtual void ProcessBlock(Block block)
+        {
+            Guard.NotNull(block, nameof(block));
+
+            this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
+
+            if (!this.walletManager.ContainsWallets)
+            {
+                this.logger.LogTrace("(-)[NO_WALLET]");
+                return;
+            }
+
+            if (block.BlockSize != null)
```

Comment:
As mentioned above, this should not be needed

Reply:
ok